### PR TITLE
A11ybugs: Fixed most of the Accesibility bugs on AMP Start

### LINKED ISSUE #373: [A11Y] Android: Sidebar Menu and close buttons are not labeled
**ENVIRONMENT:**

OS (number): N (7.1.2)
Build number: N2G47H
Form factor (Smartphone / tablet):  Nexus 6P
Browser: Chrome Dev 59.0.3056.4

**Pre-requisite:**
1. Talkback is ON

**STEPS TO REPRODUCE:**

1. Open [https://ampstart.com](https://ampstart.com)))
2. Try to Navigate to Sidebar Menu and content within it.

**ACTUAL RESULTS:**

1. When user navigates to the Menu button, talkback reads it “button, double tap to activate”. It doesn’t reads out any label for it.
2. Same issue is observed when user navigates on the close (X) button present in the sidebar Menu. Screenreader user is also provided with an invisible close Menu button additionally, which can be reached at the end after navigating through the Menu items

**EXPECTED RESULTS:**

1. Sidebar Menu button should be labeled appropriately.
2. There should be only one close button  and it should have appropriate label.

**ATTACHMENTS**

unlabeledsidebarmenu_button
unlabeled close button

### LINKED ISSUE #383: [A11Y] [iOS] Sidebar Menu and Close Button are not labelled appropriately.
**ENVIRONMENT:**
Form factor : iPAD
Model : MC705LL/A
Version : 9.3.5(13G36)
Browser : Safari 

**Pre-requisite:**
Turn 'VoiceOver' ON. ( Settings->General->Accessibility->VoiceOver->ON )

**STEPS TO REPRODUCE:**
1. Navigate to the URL : https://ampstart.com
2. Try to linear navigate to Sidebar Menu and content within it.

**ACTUAL RESULTS:**
1. When user navigates to the Menu button, VoiceOver reads some random label i.e, 'try gram per heaven button' for the Sidebar Menu.
2. Close button is labelled as 'Button'.

**EXPECTED RESULTS:**
1. Sidebar Menu button should be labelled appropriately.
2.Close button should have an appropriate label.

Please see the attached screenshots for reference.

close button
sidebar menu

### LINKED ISSUE #388: [A11Y] Android: Responsive Menubar component have hidden graphic on it
**ENVIRONMENT:**

OS (number): N (7.0)
Build number: NRD91R
Form factor (Smartphone / tablet):  Nexus 9
Browser: Chrome Dev 59.0.3062.4

**Pre-requisite:**
1. Talkback is ON

**STEPS TO REPRODUCE:**

1. Open [https://ampstart.com](https://ampstart.com)
2. Linearly Navigate to COMPONENTS link and double tap on it.
3. Linearly navigate to the “Navigation” components section and linearly navigates through the Responsive Menubar.

**ACTUAL RESULTS:**

While linearly navigating a hidden image is encountered and talkback says it as “The blog, graphic”

**EXPECTED RESULTS:**

No hidden content should be encountered or identified when using screen reader.

**ATTACHMENT(S):**

hidden graphic

### LINKED ISSUE #395: [A11Y] [iOS] Social Media Share buttons do not have appropriate labels assigned.
ENVIRONMENT:
Form factor : iPAD
Model : MC705LL/A
Version : 9.3.5(13G36)
Browser : Safari

Pre-requisite:
Turn 'VoiceOver' ON. ( Settings->General->Accessibility->VoiceOver->ON )

STEPS TO REPRODUCE:

1. Open https://ampstart.com
2. Linearly navigate to the “Article” template and double tap on View Template
3. Linearly navigate to READ MORE section and try to move focus on social media buttons using linear navigation or exploratory touch and observe what Voiceover reads for each icon label.

ACTUAL RESULTS:
Social media share buttons are labelled as "Link".

EXPECTED RESULTS:
The social media share buttons should have appropriate labels and controls assigned to it.

ATTACHMENT(S):
Please refer to the videos in the below link for reference:

https://drive.google.com/corp/drive/folders/0BwgKOOZP99EuUzVWT25TVzl2S00

### LINKED ISSUE #402: [A11Y] [iOS] Side bar Menu and Close button inside the menu are inappropriately labelled in "What is AMP?" page.
**ENVIRONMENT:**
Form factor : iPAD
Model : MC705LL/A
Version : 9.3.5(13G36)
Browser : Safari

**PRE-REQUISITE:**
Turn 'VoiceOver' ON. ( Settings->General->Accessibility->VoiceOver->ON )

**STEPS TO REPRODUCE:**

1. Navigate to the URL: https://ampstart.com/
2. Navigate to "What is AMP?" link in the side bar menu and double tap on it.
3. Linear navigate to the Side Bar Menu in this page and observe what Voice Over announces and then double tap on it.
4. Navigate to the Close button and observe what Voice Over announces.

**ACTUAL RESULTS:**
Sidebar menu is labelled as "Button and banner" and close button is labelled as "Button".

**EXPECTED RESULTS:**
Sidebar Menu and Close buttons should have appropriate labels assigned.

**ATTACHMENT:**
Please see the video in the link below for reference.

https://drive.google.com/corp/drive/folders/0BwgKOOZP99EuUzVWT25TVzl2S00

### LINKED ISSUE #409: [A11Y] [iOS] Close button in the preview banner does not have an appropriate label.
**ENVIRONMENT:**
Form factor : iPAD
Model : MC705LL/A
Version : 9.3.5(13G36)
Browser : Safari

**PRE-REQUISITE:**
Turn 'VoiceOver' ON. ( Settings->General->Accessibility->VoiceOver->ON )

**STEPS TO REPRODUCE:**

1. Navigate to the URL: https://ampstart.com/
2. Navigate to "Templates" link and double tap on it.
3. Navigate to 'VIEW TEMPLATE' in any of the sections and double tap on it.
4. Try to linear navigate to close button in the preview banner and observe what VoiceOver announces.

**ACTUAL RESULTS:**
Close button is labelled as button.

**EXPECTED RESULTS:**
Close button should have an appropriate label assigned.

**ATTACHMENT:**
Please see the video in the link below for reference.

https://drive.google.com/corp/drive/folders/0BwgKOOZP99EuUzVWT25TVzl2S00

### LINKED ISSUE #411: [A11Y] [Web] Linked Icons at bottom of Template page not properly labeled
ENVIRONMENT:
Build number: 56.0.2924.87 beta (64-bit)
Form factor: Chromebook
Browser: Chrome

PRE-REQUISITE
chromevox is ON

STEPS TO REPRODUCE:
Open https://ampstart.com
Navigate to Templates tab
Using chromevox, navigate to share icon links located at bottom of page
Navigate through icons

ACTUAL RESULTS:
Using chromevox, user can navigate to linked icons at bottom of template main landing page.  As user navigates through icons, icons read off with twitter and wordpress but github and slack have no identification of what icon is.

EXPECTED RESULTS:
Icons are labeled correctly

ATTACHMENT:
Please see the video in the link below for reference.

https://drive.google.com/open?id=0BwUNSMCBrCB4WmM3cGU5WTJvWVE

### LINKED ISSUE #413: [A11Y] [Web] Close button in preview template mode labeled incorrectly
ENVIRONMENT:
Build number: 56.0.2924.87 beta (64-bit)
Form factor: Chromebook
Browser: Chrome

PRE-REQUISITE
chromevox is ON

STEPS TO REPRODUCE:
1. Open https://ampstart.com
2. Using chromevox, to Templates tab
3. Navigate to view template button.
4. Open Template
5. Navigate to X button in pop up preview menu

ACTUAL RESULTS:
Using chromevox, navigate view templates button and select.  Pop up will appear with template sample.  Once user is navigating through template preview and selects the side list, close button for list reads out incorrectly

EXPECTED RESULTS:
X button reads out close or something similar.

ATTACHMENT:
Please see the video in the link below for reference.

https://drive.google.com/open?id=0BwUNSMCBrCB4Q3F1OVliaGh0dms

### LINKED ISSUE #414: [A11Y] [Web] Amp Start Logo incorrectly reads "the blog"
ENVIRONMENT:
Build number: 56.0.2924.87 beta (64-bit)
Form factor: Chromebook
Browser: Chrome

PRE-REQUISITE
chromevox is ON

STEPS TO REPRODUCE:
1. Open https://ampstart.com
2. Navigate to Templates tab
3. Using chromevox, navigate amp start logo at left top of screen

ACTUAL RESULTS:
Using chromevox, navigate view templates and navigate to Amp Start Logo. Logo reads “the blog”

EXPECTED RESULTS:
Logo reads Amp Start

ATTACHMENT:
Please see the video in the link below for reference.

https://drive.google.com/open?id=0BwUNSMCBrCB4MmJJZzA1czc4b2s

### LINKED ISSUE #417: [A11Y] [Web] Share Icons on Component Page not labeled
ENVIRONMENT:
Build number: 56.0.2924.87 beta (64-bit)
Form factor: Chromebook
Browser: Chrome

PRE-REQUISITE
chromevox is ON

STEPS TO REPRODUCE:
1. Open https://ampstart.com
2. Navigate to Components Tab
3. Using chromevox, navigate to Share Icons
4. Cycle through Share Icons

ACTUAL RESULTS:
Share Icons read as link

EXPECTED RESULTS:
Icon should read properly ie bird icon for twitter etc.

ATTACHMENT:

### LINKED ISSUE #418: [A11Y] [Web] Hidden Image under Navigation
ENVIRONMENT:
Build number: 56.0.2924.87 beta (64-bit)
Form factor: Chromebook
Browser: Chrome

PRE-REQUISITE
chromevox is ON

STEPS TO REPRODUCE:
1. Open https://ampstart.com
2. Navigate to Components Tab
3. Using chromevox, Navigate to Navigation header and move through Responsive Menubar options
4. Navigate to blank image left of “fashion” button 

ACTUAL RESULTS:
Hidden image reads “the blog”

EXPECTED RESULTS:
Image should be shown and properly labeled

ATTACHMENT:
https://drive.google.com/open?id=0BwUNSMCBrCB4WU14bjBacGFFUGM

### LINKED ISSUE #419: [A11Y] [Web] Get Started Tab Template images not labeled
ENVIRONMENT:
Build number: 56.0.2924.87 beta (64-bit)
Form factor: Chromebook
Browser: Chrome

PRE-REQUISITE
chromevox is ON

STEPS TO REPRODUCE:
1. Open https://ampstart.com
2. Navigate to Get Started Tab
3. Using chromevox, Navigate Step 1: Pick a Template
4. Navigate to template images

ACTUAL RESULTS:
Images read out template one or template 2

EXPECTED RESULTS:
Template Image should be shown and properly labeled

ATTACHMENT:
https://drive.google.com/open?id=0BwUNSMCBrCB4Ul9kZlBUT0lWTlE

### PR DESCRIPTION
**Pull request by torch2424**
Everything was fixed using [Chromevox](https://chrome.google.com/webstore/detail/chromevox/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en), I am closing some of the Mobile specific issue, since often times, they're were cause by the same problem.

# Bugs Fixed

closes #373 
closes #388 
closes #418 
closes #409 
closes #402 
closes #383 
closes #414 
closes #413 
closes #411 
closes #417 
closes #419 
closes #395

More may have been closed, as many are duplicated on different devices. If there are, please let me know as I am tracking this personally.

# Example Screenshot
screen shot 2017-06-12 at 11 09 56 am

### CODE REVIEW

**Comment by torch2424** [comment]
I rebased the PR to not include my @google git config, my apologies, didn't notice beforehand

File: data.json [hunk 1]
```diff
@@ -398,12 +398,6 @@
       },
       "navbar-sidebar-only": {
         "navbar": {
-          "logo": {
```
Comment (camelburrito) [inline data.json]: why is this being removed?

File: data.json [hunk 2]
```diff
@@ -470,12 +464,6 @@
       },
       "navbar-with-sidebar": {
         "navbar": {
-          "logo": {
```
Comment (camelburrito) [inline data.json]: why?

File: templates/blog.amp.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
         "width": "100",
         "height": "61.3"
       },
-      "name": "The Blog",
+      "name": "The Blog logo",
```
Comment (camelburrito) [inline templates/blog.amp.json]: This is not right

File: www/components.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
         "classes": "xs-hide sm-hide",
         "href": "/"
       },
-      "name": "The Blog",
+      "name": "AMP start home logo",
```
Comment (camelburrito) [inline www/components.json]: Ditto

File: www/getstarted.json [hunk 5]
```diff
@@ -24,7 +24,7 @@
         "classes": "xs-hide sm-hide",
         "href": "/"
       },
-      "name": "The Blog",
+      "name": "AMP start home logo",
```
Comment (camelburrito) [inline www/getstarted.json]: May be the logo should have an alt - this should not change here

File: www/howitworks.json [hunk 6]
```diff
@@ -23,7 +23,7 @@
         "classes": "xs-hide sm-hide",
         "href": "/"
       },
-      "name": "The Blog",
+      "name": "AMP start home logo",
```
Comment (camelburrito) [inline www/howitworks.json]: ditto

File: www/index.json [hunk 7]
```diff
@@ -27,7 +27,7 @@
         "classes": "xs-hide sm-hide",
         "href": "/"
       },
-      "name": "The Blog",
+      "name": "AMP start home logo",
```
Comment (camelburrito) [inline www/index.json]: ditto

File: data.json (hunk 1 above)
Comment (torch2424) [inline data.json]: This fixes the "hidden graphic" on the sidebar examples on the components page.

The example was rendering an empty logo as the example did not have one. I ensured that this did not break other sidebars across the site

See #388

File: data.json (hunk 2 above)
Comment (torch2424) [inline data.json]: This fixes the "hidden graphic" on the sidebar examples on the components page.

The example was rendering an empty logo as the example did not have one. I ensured that this did not break other sidebars across the site

See #388

File: templates/blog.amp.json (hunk 3 above)
Comment (torch2424) [inline templates/blog.amp.json]: Could you provide proper text for this then? The Templates and things usually have random logos.

File: www/components.json (hunk 4 above)
Comment (torch2424) [inline www/components.json]: Could you provide proper text for this then? The Templates and things usually have random logos.

File: www/howitworks.json (hunk 6 above)
Comment (torch2424) [inline www/howitworks.json]: Could you provide proper text for this then? The Templates and things usually have random logos.

File: www/index.json (hunk 7 above)
Comment (torch2424) [inline www/index.json]: Could you provide proper text for this then? The Templates and things usually have random logos.

File: www/getstarted.json (hunk 5 above)
Comment (torch2424) [inline www/getstarted.json]: I agree, that would be a good idea :)

I can implement this now

Comment (torch2424) [inline www/getstarted.json]: We agreed, that the the text for this is correct. This shall stay the same

File: templates/blog.amp.json (hunk 3 above)
Comment (torch2424) [inline templates/blog.amp.json]: We agreed that the "The Blog" changes were correct

File: components/navbar/navbar.snip.html [hunk 8]
```diff
@@ -19,7 +19,7 @@
 {{#navbar}}
   <!-- Start Navbar -->
   <header class="ampstart-headerbar fixed flex justify-start items-center top-0 left-0 right-0 pl2 pr4">
-    <div role="button" on="tap:{{sidebar-id}}.toggle" tabindex="0" class="ampstart-navbar-trigger {{^sidebaronly}}md-hide lg-hide{{/sidebaronly}} pr2">☰</div>
+    <div role="button" aria-label="open sidebar navigation" on="tap:{{sidebar-id}}.toggle" tabindex="0" class="ampstart-navbar-trigger {{^sidebaronly}}md-hide lg-hide{{/sidebaronly}} pr2">☰</div>
```
Comment (camelburrito) [inline components/navbar/navbar.snip.html]: "Open Sidebar" should do?

File: components/social-follow/social-follow.snip.html [hunk 9]
```diff
@@ -22,31 +22,31 @@
   {{#links}}
     <li class="mr2">
       {{#twitter}}
-        <a href="{{{.}}}" target="_blank" class="inline-block">{{> ../../img/icons/twitter.svg}}</a>
+        <a href="{{{.}}}" target="_blank" class="inline-block" aria-label="Twitter icon link">{{> ../../img/icons/twitter.svg}}</a>
```
Comment (camelburrito) [inline components/social-follow/social-follow.snip.html]: "Link to our Twitter page"

File: components/social-follow/social-follow.snip.html [hunk 10]
```diff
@@ -22,31 +22,31 @@
...
-        <a href="{{{.}}}" target="_blank" class="inline-block">{{> ../../img/icons/github.svg}}</a>
+        <a href="{{{.}}}" target="_blank" class="inline-block" aria-label="Github icon link">{{> ../../img/icons/github.svg}}</a>
       {{/github}}
       {{#wordpress}}
-        <a href="{{{.}}}" target="_blank" class="inline-block">{{> ../../img/icons/wordpress.svg}}</a>
+        <a href="{{{.}}}" target="_blank" class="inline-block" aria-label="Wordpress icon link">{{> ../../img/icons/wordpress.svg}}</a>
       {{/wordpress}}
       {{#slack}}
-        <a href="{{{.}}}" target="_blank" class="inline-block">{{> ../../img/icons/slack.svg}}</a>
+        <a href="{{{.}}}" target="_blank" class="inline-block" aria-label="Slack icon link">{{> ../../img/icons/slack.svg}}</a>
```
Comment (camelburrito) [inline components/social-follow/social-follow.snip.html]: Link to our Slack channel

File: data.json [hunk 11]
```diff
@@ -471,11 +471,11 @@
       "navbar-with-sidebar": {
         "navbar": {
           "logo": {
-            "url": "../img/thescenic/header-logo.png",
+            "url": "../img/components/sidebar-example-logo.png",
```
Comment (camelburrito) [inline data.json]: you will need to updated these in the appropriate files in the `hl-partials` folder

File: templates/blog.amp.json (hunk 3 above)
Comment (camelburrito) [inline templates/blog.amp.json]: This should not change in for the blog template right?

File: data.json [hunk 12]
```diff
@@ -399,11 +399,12 @@
...
+            "alt": "Sidebar blog logo",
             "width": "110",
             "height": "33"
           },
-          "name": "The Blog",
+          "name": "The blog logo",
```
Comment (camelburrito) [inline data.json]: This should not change?

File: data.json [hunk 13]
```diff
@@ -399,11 +399,12 @@
       "navbar-sidebar-only": {
         "navbar": {
           "logo": {
-            "url": "../img/thescenic/header-logo.png",
+            "url": "../img/components/sidebar-example-logo.png",
+            "alt": "Sidebar blog logo",
```
Comment (camelburrito) [inline data.json]: Example Logo image

File: data.json (hunk 11 above)
Comment (camelburrito) [inline data.json]: is this done?

File: templates/blog.amp.json (hunk 3 above)
Comment (camelburrito) [inline templates/blog.amp.json]: This should not have changed.

File: www/components.json (hunk 4 above)
Comment (camelburrito) [inline www/components.json]: Same here ?

File: www/getstarted.json (hunk 5 above)
Comment (camelburrito) [inline www/getstarted.json]: I dont seem to see your changes in the JSON file - did you forget to push?

Comment (camelburrito) [inline www/getstarted.json]: This should have been on alt property that you were supposed to add to the logo right?

File: data.json (hunk 11 above)
Comment (torch2424) [inline data.json]: Yes this is done, please the example screenshot I attached 😄

**Comment by torch2424** [comment]
Commenting to get cla/google to run

**Pull request merged**